test/parallel/test-crypto-pbkdf2.js

```diff
@@ -63,33 +63,17 @@ common.expectsError(
   }
 );
 
-// Should not work with Infinity key length
-assert.throws(function() {
-  crypto.pbkdf2('password', 'salt', 1, Infinity, 'sha256',
-                common.mustNotCall());
-}, /^TypeError: Bad key length$/);
-
-// Should not work with negative Infinity key length
-assert.throws(function() {
-  crypto.pbkdf2('password', 'salt', 1, -Infinity, 'sha256',
-                common.mustNotCall());
-}, /^TypeError: Bad key length$/);
-
-// Should not work with NaN key length
-assert.throws(function() {
-  crypto.pbkdf2('password', 'salt', 1, NaN, 'sha256', common.mustNotCall());
-}, /^TypeError: Bad key length$/);
-
-// Should not work with negative key length
-assert.throws(function() {
-  crypto.pbkdf2('password', 'salt', 1, -1, 'sha256', common.mustNotCall());
-}, /^TypeError: Bad key length$/);
-
-// Should not work with key length that does not fit into 32 signed bits
-assert.throws(function() {
-  crypto.pbkdf2('password', 'salt', 1, 4073741824, 'sha256',
-                common.mustNotCall());
-}, /^TypeError: Bad key length$/);
+[Infinity, -Infinity, NaN, -1, 4073741824].forEach((i) => {
```

Comment:
Just a dumb question: Till today my understanding was that `INT_MAX` would be the size of the word of the given processor architecture. But today, even on my 64-bit machine, the value is defined as `2147483647`. I am a little confused here. C++ Spec is not very clear in this regard. Do you know where I can get a clear answer?

Reply:
@thefourtheye `INT_MAX` is the maximal value of the signed `int` type. `int` can be anything larger than or equal to 16 bits, although most compilers implement it as 32 bits. The number you are seeing is `2^31-1`.

Reply:
@tniessen That is exactly the problem. There is no guarantee that it is going to be always `2^31-1`, right? Otherwise this test will fail.

Reply:
@thefourtheye I believe all compilers we officially support use 32 bit `int`s on all supported platforms, though I might be mistaken. If I remember correctly, there is also `INT32_MAX`, but I am not sure it is more appropriate to use. `INT_MAX` seems fine as long as we use `int` internally. If `INT_MAX` is greater on some platforms, e.g. `2^63-1`, it will likely be bigger than `MAX_SAFE_INTEGER` (which I believe is around `2^54`).

Reply:
Ah, "all compilers we officially support"- that is the key here. Let me start a CI to confirm this.

Reply:
If just to be safe, we can use `process.binding('constants').crypto.INT_MAX + 1` here